src/utils/curriculum/getNumberOfSelectedUnits.ts

```diff
@@ -0,0 +1,26 @@
+import { isVisibleUnit } from "./isVisibleUnit";
+
+// TODO: These types should be moved out of components
+import {
+  YearData,
+  YearSelection,
+} from "@/components/CurriculumComponents/CurriculumVisualiser";
+
+export function getNumberOfSelectedUnits(
+  yearData: YearData,
+  selectedYear: string | null,
+  yearSelection: YearSelection,
+): number {
+  let count = 0;
+
+  Object.keys(yearData).forEach((year) => {
+    const units = yearData[year]?.units;
+    if (units && (selectedYear === "" || selectedYear === year)) {
+      units.forEach((unit) => {
+        isVisibleUnit(yearSelection, year, unit) ? count++ : count;
```

Comment:
This is a little odd, hence the [sonar cloud warning](https://sonarcloud.io/project/issues?severities=BLOCKER%2CCRITICAL%2CMAJOR%2CMINOR&sinceLeakPeriod=true&issueStatuses=OPEN%2CCONFIRMED&types=BUG&pullRequest=2785&id=oaknational_owa&open=AZICpe558EdTur4Z4FPb)  

Clearer would be

```js
count = isVisibleUnit(yearSelection, year, unit) ? count+1 : count
```

Reply:
To increment in ternary form requires the `++` operator I believe – as Sonar is flagging this I've reverted it back it to it's previous if statement form and used the `+=1` assignment